test/unit/BPool/BPool_Bind.tree

```diff
@@ -0,0 +1,26 @@
+BPool::Bind
+├── when reentrancy lock is set
+│   └── it should revert
+├── when caller is NOT controller
+│   └── it should revert
+└── when caller is controller
+    ├── when token is already bound
+    │   └── it should revert
+    ├── when pool is finalized
+    │   └── it should revert
+    ├── when MAX_BOUND_TOKENS tokens are already bound
+    │   └── it should revert
+    ├── when token weight is too low
+    │   └── it should revert
+    ├── when token weight is too high
+    │   └── it should revert
+    ├── when too little balance is provided
+    │   └── it should revert
+    ├── when weight sum exceeds MAX_TOTAL_WEIGHT
+    │   └── it should revert
+    └── when token can be bound
+        ├── it sets reentrancy lock
```

Comment:
this one will be hard to test, am i right?

Reply:
I like `.tree` files to be less code-centric, and not have to change this if we manage reentrancy locks some other way, similar to what you described regarding the swapping and popping of the `_tokens` array in unbind